ci/scripts/run-codebuild/libs/codebuild-runner.test.js

```diff
@@ -0,0 +1,250 @@
+import { CodeBuildRunner, CodeBuildError } from './codebuild-runner.js'
+
+import { mockClient } from 'aws-sdk-client-mock'
+import { CloudWatchLogsClient, GetLogEventsCommand } from '@aws-sdk/client-cloudwatch-logs'
+import { CodeBuildClient, StartBuildCommand, BatchGetBuildsCommand } from '@aws-sdk/client-codebuild'
+
+const cbMock = mockClient(CodeBuildClient)
+const cwlMock = mockClient(CloudWatchLogsClient)
+
+// Note these mock responses only include what is actually being used to reduce the size of them
+const mockResponses = {
+  startBuildCommandResponse: {
+    build: {
+      id: 'mockBuildId',
+      buildNumber: 123,
+      projectName: 'mockProject',
+      startTime: 'mockStartTime'
+    }
+  },
+  batchGetBuildsCommandResponse: {
+    builds: [{
+      buildComplete: false,
+      buildStatus: 'IN_PROGRESS',
+      logs: {
+        groupName: 'mockBuildLogGroupName',
+        streamName: 'mockBuildLogStreamName',
+        deepLink: 'mockBuildLogDeepLink'
+      }
+    }]
+  },
+  getLogEventsCommandResponses: [
+    {
+      events: [
+        { message: 'mock message 1 event 1' },
+        { message: 'mock message 1 event 2' }
+      ],
+      nextForwardToken: 'mockForwardToken1'
+    },
+    {
+      events: [
+        { message: 'mock message 2 event 1' },
+        { message: 'mock message 2 event 2' }
+      ],
+      nextForwardToken: 'mockForwardToken2'
+    },
+    {
+      events: [
+        { message: 'mock message 3 event 1' },
+        { message: 'mock message 3 event 2' }
+      ],
+      nextForwardToken: 'mockForwardToken3'
+    },
+    {
+      events: [
+        { message: 'mock message 4 event 1' },
+        { message: 'mock message 4 event 2' }
+      ],
+      // Note this token is the same as the previous to indicate the end of the log stream
+      nextForwardToken: 'mockForwardToken3'
+    }
+  ]
+}
+
+describe('CodeBuildRunner', () => {
+  beforeEach(() => {
+    process.stdout.write = jest.fn()
+    cbMock.reset()
+    cwlMock.reset()
+    cwlMock.onAnyCommand().rejects('CLOUDWATCH CALL')
```

Comment:
Checking my understanding: does this line stub out everything by default, then below we return responses for specific commands?

Reply:
That's correct

Reply:
Well sort of correct. It's a mock already which I believe will return undefined for anything not actively stubbed. So I'm adding this to make sure if there any unexpected calls to cloudwatch it will error